### DISCUSSION TITLE
Support for intel embedded CPUs

### DISCUSSION THREAD

**Discussion by giuliano-97** [larq/compute-engine#674] [Ideas]
Hello

I have been looking around for a library to run binary neural networks on some low-end robots like the NAO V6 from SoftBank and I came across larq. Unfortunately, it currently lacks support for Intel CPUs (I think?).

Do you plan to add support for Intel CPUs [like this one](https://ark.intel.com/content/www/us/en/ark/products/78475/intel-atom-processor-e3845-2m-cache-1-91-ghz.html) in the future?

If not, could you roughly estimate the development effort? Would you  be open to PRs for this kind of feature or do you think it is not worth maintaing?

**Comment by Tombana** [comment]
Hi @giuliano-97,

You're right that we currently don't have optimized kernels for the x64 instruction set (which is what those CPUs use), so it would run on slower fallback code.

One thing you can try is to apply the suggestion that @AdamHillier made at the end of his post here: https://github.com/larq/compute-engine/discussions/658 , i.e. enabling our experimental kernels by adding `Register_BCONV_2D_OPT_INDIRECT_BGEMM();`
Even though they were also intended for 64-bit ARM, they are likely still faster than the regular kernels on x64.

> If not, could you roughly estimate the development effort? Would you be open to PRs for this kind of feature or do you think it is not worth maintaing?

We would happily welcome these kind of PRs. Here is an example of a community PR where someone added ARM32 support: https://github.com/larq/compute-engine/pull/432/files
